### PR TITLE
Add an endpoint that gets available tables for reservation

### LINKED ISSUE #92: Create API function to get available reservation times for a restaurant
**User Story:**

As a user, I want to see all the times that bookings are available so that I can make a reservation. 

**Acceptance Criteria:**
- [ ] Created a function that gets available times of restaurants - considering restaurant table availability

### PR DESCRIPTION
**Pull request by nidhineshnand**
## Link to Issue Number
Closes #92 

## Description
Add a call to get all available tables for a restaurant.
Added unit test for feature

## Checklist
- [x] All feature request A/C have been met OR the bug has been fixed
- [x] I ran all necessary tests and they pass
- [x] I rebased upstream/master onto my working branch before opening this PR
- [x] I have named my PR something sensible
- [x] I have included the issue number above
- [x] I have assigned atleast two people to review my changes

### CODE REVIEW

File: backend/routes/reservation.js [hunk 1]
```diff
@@ -137,4 +137,36 @@ router.delete('/single', async (req, res) => {
...
+      error:
+        'reservation/available GET endpoint needs: date, time, numberOfGuests and restaurantID body params'
+    });
+    return;
+  }
+
+  const { error, result } = await connection.asyncQuery(
+    'SELECT * ' +
+    'FROM restaurant_db.TABLE t ' +
+    'WHERE t.RestaurantID = ? AND t.maxGuests >= ? AND NOT EXISTS ( SELECT * ' +
```
Comment (jacoballen4534) [inline backend/routes/reservation.js]: Are we ditching the minGuests? I am unsure what is for.

Comment (nidhineshnand) [reply backend/routes/reservation.js]: It seems like a strange thing to check but I can add it

Comment (nidhineshnand) [reply backend/routes/reservation.js]: Done

File: backend/routes/reservation.js [hunk 2]
```diff
@@ -208,4 +208,65 @@ router.delete('/single', async (req, res) => {
...
+ *
+ * /reservation:
+ *   get:
+ *     description: Get a list of tables that are free for booking
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - name: restaurantID
+ *         description: Primary Key of Restaurant database table
+ *         in: query
```
Comment (wshin387) [inline backend/routes/reservation.js]: in request body (formData) not query.

Comment (nidhineshnand) [reply backend/routes/reservation.js]: Done

File: backend/routes/reservation.js [hunk 3]
```diff
@@ -208,4 +208,68 @@ router.delete('/single', async (req, res) => {
   res.json({ result: 'Deleted reservation' });
 });
 
+/**
+ * @swagger
+ *
+ * /reservation:
```
Comment (jacoballen4534) [inline backend/routes/reservation.js]: Should be /available

File: backend/routes/reservation.js [hunk 4]
```diff
@@ -208,4 +208,68 @@ router.delete('/single', async (req, res) => {
...
+ *       - name: time
+ *         description: The start time of the booking
+ *         in: formData
+ *         required: true
+ *         type: string
+ *       - name: date
+ *         description: Date for when the booking is made
+ *         in: formData
+ *         required: true
+ *         type: date
```
Comment (jacoballen4534) [inline backend/routes/reservation.js]: I believe this needs to be ```type: string``` to be able to test in the browser.

**Comment by ljcnz** [comment]
Nitpicky and perhaps not worth the effort, but was there any reason this wasn't just added ontop of the existing GET /restaurant/all api with a query param flagging to only return available bookings?

**Comment by nidhineshnand** [comment]
@jacoballen4534 Seems like swagger does not support body for GET requests.

File: backend/routes/reservation.js (hunk 3 above)
Comment (nidhineshnand) [reply backend/routes/reservation.js]: Done

File: backend/routes/reservation.js (hunk 4 above)
Comment (nidhineshnand) [reply backend/routes/reservation.js]: Done

**Comment by jacoballen4534** [comment]
> @jacoballen4534 Seems like swagger does not support body for GET requests.

Do you think it's worth changing the request type or putting params in URL, or just accept that we can't test it in swagger?

**Comment by ljcnz** [comment]
> > @jacoballen4534 Seems like swagger does not support body for GET requests.
> 
> Do you think it's worth changing the request type or putting params in URL, or just accept that we can't test it in swagger?

Yeah get requests shouldn't have anything in the post body, just move them to query params.

**Pull request merged**